debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java

```diff
@@ -606,44 +675,49 @@ public void raiseError(Exception e) {
                     }
                 } catch (Throwable t) {
                     fail("Error while trying to run connector class '" + connectorClassName + "'", t);
-                    return;
                 } finally {
                     // Close the offset storage and finally the connector ...
                     try {
                         offsetStore.stop();
                     } finally {
                         connector.stop();
+                        connectorCallback.ifPresent(ConnectorCallback::connectorStopped);
```

Comment:
don't know....the current callbacks only get called for non-exceptional cases
Currently if `connector.stop` or `task.stop` fails, no `ConnectorCallback` methods get called. If we want to handle these cases as well, then we could add an optional `Throwable` parameter to the callbacks for example.
That being said, if we do that we should also do the same for the `start` methods (which can also fail). 
So I'm not convinced; maybe the current `CompletionCallback` is enough to handle all exceptional cases.